Daily acceptance test failed: VPCRouter.Update

VPCRouterのUpdateでエラーになった場合のエラーメッセージ組み立てでnil pointer dereferenceが発生する。

該当箇所:

```go
	vpcRouter, err = builder.Update(ctx, zone, vpcRouter.ID)
	if err != nil {
		// **エラー発生箇所** vpcRouterはnilになり得る
		return fmt.Errorf("updating SakuraCloud VPCRouter[%s] is failed: %s", vpcRouter.ID, err)
	}
```


Stack trace:

```console
panic: runtime error: invalid memory address or nil pointer dereference
  [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12b7cb6]
  goroutine 9235 [running]:
  github.com/sacloud/terraform-provider-sakuracloud/sakuracloud.resourceSakuraCloudVPCRouterUpdate(0xc001c9c3f0, 0x14c7200, 0xc000f94fa0, 0x0, 0x0)
        /opt/buildagent/work/fea88cb48f140ea/sakuracloud/resource_sakuracloud_vpc_router.go:560 +0x626
  github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0002acc80, 0xc00023bdb0, 0xc0009a9fe0, 0x14c7200, 0xc000f94fa0, 0xc002683301, 0xc0024ac330, 0xc0026833d0)
        /opt/buildagent/work/fea88cb48f140ea/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:311 +0x263
  github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc0002acd00, 0xc001e07918, 0xc00023bdb0, 0xc0009a9fe0, 0xc0027a95d8, 0xc001665f01, 0x14b5600)
        /opt/buildagent/work/fea88cb48f140ea/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x99
  github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc000b809a8, 0x1aac4e0, 0xc00276cab0, 0xc000adfda0, 0xc000b809a8, 0x0, 0xc000d9ea80)
        /opt/buildagent/work/fea88cb48f140ea/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:885 +0x882
  github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x16a4400, 0xc000b809a8, 0x1aac4e0, 0xc00276cab0, 0xc000adfd40, 0x0, 0x1aac4e0, 0xc00276cab0, 0xc002776000, 0xaad)
        /opt/buildagent/work/fea88cb48f140ea/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3189 +0x217
  google.golang.org/grpc.(*Server).processUnaryRPC(0xc001e4c2c0, 0x1ac1e80, 0xc001961380, 0xc002295500, 0xc000db3d70, 0x2560aa0, 0x0, 0x0, 0x0)
        /opt/buildagent/work/fea88cb48f140ea/vendor/google.golang.org/grpc/server.go:995 +0x460
  google.golang.org/grpc.(*Server).handleStream(0xc001e4c2c0, 0x1ac1e80, 0xc001961380, 0xc002295500, 0x0)
        /opt/buildagent/work/fea88cb48f140ea/vendor/google.golang.org/grpc/server.go:1275 +0xd97
  google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000642d70, 0xc001e4c2c0, 0x1ac1e80, 0xc001961380, 0xc002295500)
        /opt/buildagent/work/fea88cb48f140ea/vendor/google.golang.org/grpc/server.go:710 +0xbb
  created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/buildagent/work/fea88cb48f140ea/vendor/google.golang.org/grpc/server.go:708 +0xa1
```

--- a/sakuracloud/resource_sakuracloud_archive.go
+++ b/sakuracloud/resource_sakuracloud_archive.go
@@ -179,9 +179,8 @@ func resourceSakuraCloudArchiveUpdate(d *schema.ResourceData, meta interface{}) 
 		return fmt.Errorf("could not read SakuraCloud Archive[%s]: %s", d.Id(), err)
 	}
 
-	archive, err = archiveOp.Update(ctx, zone, archive.ID, expandArchiveUpdateRequest(d))
-	if err != nil {
-		return fmt.Errorf("updating SakuraCloud Archive[%s] is failed: %s", archive.ID, err)
+	if _, err = archiveOp.Update(ctx, zone, archive.ID, expandArchiveUpdateRequest(d)); err != nil {
+		return fmt.Errorf("updating SakuraCloud Archive[%s] is failed: %s", d.Id(), err)
 	}
 
 	return resourceSakuraCloudArchiveRead(d, meta)
@@ -207,7 +206,7 @@ func resourceSakuraCloudArchiveDelete(d *schema.ResourceData, meta interface{}) 
 	}
 
 	if err := archiveOp.Delete(ctx, zone, archive.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud Archive[%s] is failed: %s", archive.ID, err)
+		return fmt.Errorf("deleting SakuraCloud Archive[%s] is failed: %s", d.Id(), err)
 	}
 
 	d.SetId("")

--- a/sakuracloud/resource_sakuracloud_archive_share.go
+++ b/sakuracloud/resource_sakuracloud_archive_share.go
@@ -138,7 +138,7 @@ func resourceSakuraCloudArchiveShareDelete(d *schema.ResourceData, meta interfac
 	}
 
 	if err := archiveOp.CloseFTP(ctx, zone, archive.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud Archive Share[%s] is failed: %s", archive.ID, err)
+		return fmt.Errorf("deleting SakuraCloud Archive Share[%s] is failed: %s", d.Id(), err)
 	}
 
 	d.SetId("")

--- a/sakuracloud/resource_sakuracloud_auto_backup.go
+++ b/sakuracloud/resource_sakuracloud_auto_backup.go
@@ -139,9 +139,8 @@ func resourceSakuraCloudAutoBackupUpdate(d *schema.ResourceData, meta interface{
 		return err
 	}
 
-	autoBackup, err = autoBackupOp.Update(ctx, zone, autoBackup.ID, expandAutoBackupUpdateRequest(d, autoBackup))
-	if err != nil {
-		return fmt.Errorf("updating SakuraCloud AutoBackup[%s] is failed: %s", autoBackup.ID, err)
+	if _, err = autoBackupOp.Update(ctx, zone, autoBackup.ID, expandAutoBackupUpdateRequest(d, autoBackup)); err != nil {
+		return fmt.Errorf("updating SakuraCloud AutoBackup[%s] is failed: %s", d.Id(), err)
 	}
 
 	return resourceSakuraCloudAutoBackupRead(d, meta)
@@ -167,7 +166,7 @@ func resourceSakuraCloudAutoBackupDelete(d *schema.ResourceData, meta interface{
 	}
 
 	if err := autoBackupOp.Delete(ctx, zone, autoBackup.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud AutoBackup[%s] is failed: %s", autoBackup.ID, err)
+		return fmt.Errorf("deleting SakuraCloud AutoBackup[%s] is failed: %s", d.Id(), err)
 	}
 
 	d.SetId("")

--- a/sakuracloud/resource_sakuracloud_bridge.go
+++ b/sakuracloud/resource_sakuracloud_bridge.go
@@ -107,12 +107,12 @@ func resourceSakuraCloudBridgeUpdate(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("could not read SakuraCloud Bridge[%s]: %s", d.Id(), err)
 	}
 
-	bridge, err = bridgeOp.Update(ctx, zone, bridge.ID, &sacloud.BridgeUpdateRequest{
+	_, err = bridgeOp.Update(ctx, zone, bridge.ID, &sacloud.BridgeUpdateRequest{
 		Name:        d.Get("name").(string),
 		Description: d.Get("description").(string),
 	})
 	if err != nil {
-		return fmt.Errorf("updating SakuraCloud Bridge[%s] is failed: %s", bridge.ID, err)
+		return fmt.Errorf("updating SakuraCloud Bridge[%s] is failed: %s", d.Id(), err)
 	}
 	return resourceSakuraCloudBridgeRead(d, meta)
 }

--- a/sakuracloud/resource_sakuracloud_container_registry.go
+++ b/sakuracloud/resource_sakuracloud_container_registry.go
@@ -164,7 +164,7 @@ func resourceSakuraCloudContainerRegistryUpdate(d *schema.ResourceData, meta int
 
 	builder := expandContainerRegistryBuilder(d, client, reg.SettingsHash)
 	if _, err := builder.Update(ctx, reg.ID); err != nil {
-		return fmt.Errorf("updating SakuraCloud ContainerRegistry[%s] is failed: %s", reg.ID, err)
+		return fmt.Errorf("updating SakuraCloud ContainerRegistry[%s] is failed: %s", d.Id(), err)
 	}
 
 	return resourceSakuraCloudContainerRegistryRead(d, meta)
@@ -189,7 +189,7 @@ func resourceSakuraCloudContainerRegistryDelete(d *schema.ResourceData, meta int
 	}
 
 	if err := regOp.Delete(ctx, reg.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud ContainerRegistry[%s] is failed: %s", reg.ID, err)
+		return fmt.Errorf("deleting SakuraCloud ContainerRegistry[%s] is failed: %s", d.Id(), err)
 	}
 
 	d.SetId("")

--- a/sakuracloud/resource_sakuracloud_database.go
+++ b/sakuracloud/resource_sakuracloud_database.go
@@ -235,7 +235,7 @@ func resourceSakuraCloudDatabaseUpdate(d *schema.ResourceData, meta interface{})
 
 	dbBuilder := expandDatabaseBuilder(d, client)
 	if _, err := dbBuilder.Update(ctx, zone, db.ID); err != nil {
-		return fmt.Errorf("updating SakuraCloud Database[%s] is failed: %s", db.ID, err)
+		return fmt.Errorf("updating SakuraCloud Database[%s] is failed: %s", d.Id(), err)
 	}
 
 	return resourceSakuraCloudDatabaseRead(d, meta)
@@ -268,7 +268,7 @@ func resourceSakuraCloudDatabaseDelete(d *schema.ResourceData, meta interface{})
 
 	// delete
 	if err = dbOp.Delete(ctx, zone, data.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud Database[%s] is failed: %s", data.ID, err)
+		return fmt.Errorf("deleting SakuraCloud Database[%s] is failed: %s", d.Id(), err)
 	}
 
 	d.SetId("")

--- a/sakuracloud/resource_sakuracloud_database_read_replica.go
+++ b/sakuracloud/resource_sakuracloud_database_read_replica.go
@@ -183,7 +183,7 @@ func resourceSakuraCloudDatabaseReadReplicaUpdate(d *schema.ResourceData, meta i
 
 	db, err = builder.Update(ctx, zone, db.ID)
 	if err != nil {
-		return fmt.Errorf("updating SakuraCloud Database ReadReplica[%s] is failed: %s", db.ID, err)
+		return fmt.Errorf("updating SakuraCloud Database ReadReplica[%s] is failed: %s", d.Id(), err)
 	}
 
 	return setDatabaseReadReplicaResourceData(ctx, d, client, db)
@@ -217,7 +217,7 @@ func resourceSakuraCloudDatabaseReadReplicaDelete(d *schema.ResourceData, meta i
 
 	// delete
 	if err = dbOp.Delete(ctx, zone, data.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud Database[%s] is failed: %s", data.ID, err)
+		return fmt.Errorf("deleting SakuraCloud Database[%s] is failed: %s", d.Id(), err)
 	}
 
 	d.SetId("")

--- a/sakuracloud/resource_sakuracloud_esme.go
+++ b/sakuracloud/resource_sakuracloud_esme.go
@@ -117,9 +117,8 @@ func resourceSakuraCloudESMEUpdate(d *schema.ResourceData, meta interface{}) err
 		return err
 	}
 
-	esme, err = esmeOp.Update(ctx, esme.ID, expandESMEUpdateRequest(d, esme))
-	if err != nil {
-		return fmt.Errorf("updating SakuraCloud ESME[%s] is failed: %s", esme.ID, err)
+	if _, err = esmeOp.Update(ctx, esme.ID, expandESMEUpdateRequest(d, esme)); err != nil {
+		return fmt.Errorf("updating SakuraCloud ESME[%s] is failed: %s", d.Id(), err)
 	}
 
 	return resourceSakuraCloudESMERead(d, meta)
@@ -145,7 +144,7 @@ func resourceSakuraCloudESMEDelete(d *schema.ResourceData, meta interface{}) err
 	}
 
 	if err := esmeOp.Delete(ctx, esme.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud ESME[%s] is failed: %s", esme.ID, err)
+		return fmt.Errorf("deleting SakuraCloud ESME[%s] is failed: %s", d.Id(), err)
 	}
 
 	d.SetId("")

--- a/sakuracloud/resource_sakuracloud_local_router.go
+++ b/sakuracloud/resource_sakuracloud_local_router.go
@@ -235,9 +235,8 @@ func resourceSakuraCloudLocalRouterUpdate(d *schema.ResourceData, meta interface
 		return fmt.Errorf("validating parameter for SakuraCloud LocalRouter is failed: %s", err)
 	}
 
-	localRouter, err = builder.Update(ctx, localRouter.ID)
-	if err != nil {
-		return fmt.Errorf("updating SakuraCloud LocalRouter[%s] is failed: %s", localRouter.ID, err)
+	if _, err = builder.Update(ctx, localRouter.ID); err != nil {
+		return fmt.Errorf("updating SakuraCloud LocalRouter[%s] is failed: %s", d.Id(), err)
 	}
 	return resourceSakuraCloudLocalRouterRead(d, meta)
 }
@@ -265,7 +264,7 @@ func resourceSakuraCloudLocalRouterDelete(d *schema.ResourceData, meta interface
 	}
 
 	if err := lrOp.Delete(ctx, localRouter.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud LocalRouter[%s] is failed: %s", localRouter.ID, err)
+		return fmt.Errorf("deleting SakuraCloud LocalRouter[%s] is failed: %s", d.Id(), err)
 	}
 	return nil
 }

--- a/sakuracloud/resource_sakuracloud_mobile_gateway.go
+++ b/sakuracloud/resource_sakuracloud_mobile_gateway.go
@@ -282,9 +282,8 @@ func resourceSakuraCloudMobileGatewayUpdate(d *schema.ResourceData, meta interfa
 		return fmt.Errorf("validating SakuraCloud MobileGateway is failed: %s", err)
 	}
 
-	mgw, err = builder.Update(ctx, zone, mgw.ID)
-	if err != nil {
-		return fmt.Errorf("updating SakuraCloud MobileGateway[%s] is failed: %s", mgw.ID, err)
+	if _, err = builder.Update(ctx, zone, mgw.ID); err != nil {
+		return fmt.Errorf("updating SakuraCloud MobileGateway[%s] is failed: %s", d.Id(), err)
 	}
 
 	return resourceSakuraCloudMobileGatewayRead(d, meta)
@@ -311,7 +310,7 @@ func resourceSakuraCloudMobileGatewayDelete(d *schema.ResourceData, meta interfa
 	}
 
 	if err := cleanup.DeleteMobileGateway(ctx, mgwOp, simOp, zone, mgw.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud MobileGateway[%s] is failed: %s", mgw.ID, err)
+		return fmt.Errorf("deleting SakuraCloud MobileGateway[%s] is failed: %s", d.Id(), err)
 	}
 	return nil
 }

--- a/sakuracloud/resource_sakuracloud_sim.go
+++ b/sakuracloud/resource_sakuracloud_sim.go
@@ -166,12 +166,12 @@ func resourceSakuraCloudSIMUpdate(d *schema.ResourceData, meta interface{}) erro
 
 	builder := expandSIMBuilder(d, client)
 	if err := builder.Validate(ctx); err != nil {
-		return fmt.Errorf("validating SakuraCloud SIM[%s] is failed: %s", sim.ID, err)
+		return fmt.Errorf("validating SakuraCloud SIM[%s] is failed: %s", d.Id(), err)
 	}
 
 	_, err = builder.Update(ctx, sim.ID)
 	if err != nil {
-		return fmt.Errorf("updating SakuraCloud SIM[%s] is failed: %s", sim.ID, err)
+		return fmt.Errorf("updating SakuraCloud SIM[%s] is failed: %s", d.Id(), err)
 	}
 
 	return resourceSakuraCloudSIMRead(d, meta)

--- a/sakuracloud/resource_sakuracloud_simple_monitor.go
+++ b/sakuracloud/resource_sakuracloud_simple_monitor.go
@@ -253,9 +253,8 @@ func resourceSakuraCloudSimpleMonitorUpdate(d *schema.ResourceData, meta interfa
 		return fmt.Errorf("could not read SimpleMonitor[%s]: %s", d.Id(), err)
 	}
 
-	simpleMonitor, err = smOp.Update(ctx, simpleMonitor.ID, expandSimpleMonitorUpdateRequest(d))
-	if err != nil {
-		return fmt.Errorf("updating SimpleMonitor[%s] is failed: %s", simpleMonitor.ID, err)
+	if _, err = smOp.Update(ctx, simpleMonitor.ID, expandSimpleMonitorUpdateRequest(d)); err != nil {
+		return fmt.Errorf("updating SimpleMonitor[%s] is failed: %s", d.Id(), err)
 	}
 
 	return resourceSakuraCloudSimpleMonitorRead(d, meta)
@@ -281,7 +280,7 @@ func resourceSakuraCloudSimpleMonitorDelete(d *schema.ResourceData, meta interfa
 	}
 
 	if err := smOp.Delete(ctx, simpleMonitor.ID); err != nil {
-		return fmt.Errorf("deleting SimpleMonitor[%s] is failed: %s", simpleMonitor.ID, err)
+		return fmt.Errorf("deleting SimpleMonitor[%s] is failed: %s", d.Id(), err)
 	}
 	return nil
 }

--- a/sakuracloud/resource_sakuracloud_ssh_key.go
+++ b/sakuracloud/resource_sakuracloud_ssh_key.go
@@ -115,7 +115,7 @@ func resourceSakuraCloudSSHKeyUpdate(d *schema.ResourceData, meta interface{}) e
 
 	_, err = sshKeyOp.Update(ctx, key.ID, expandSSHKeyUpdateRequest(d))
 	if err != nil {
-		return fmt.Errorf("updating SSHKey[%s] is failed: %s", key.ID, err)
+		return fmt.Errorf("updating SSHKey[%s] is failed: %s", d.Id(), err)
 	}
 	return resourceSakuraCloudSSHKeyRead(d, meta)
 }
@@ -140,7 +140,7 @@ func resourceSakuraCloudSSHKeyDelete(d *schema.ResourceData, meta interface{}) e
 	}
 
 	if err := sshKeyOp.Delete(ctx, key.ID); err != nil {
-		return fmt.Errorf("deleting SSHKey[%s] is failed: %s", key.ID, err)
+		return fmt.Errorf("deleting SSHKey[%s] is failed: %s", d.Id(), err)
 	}
 	return nil
 }

--- a/sakuracloud/resource_sakuracloud_ssh_key_gen.go
+++ b/sakuracloud/resource_sakuracloud_ssh_key_gen.go
@@ -143,7 +143,7 @@ func resourceSakuraCloudSSHKeyGenDelete(d *schema.ResourceData, meta interface{}
 	}
 
 	if err := sshKeyOp.Delete(ctx, key.ID); err != nil {
-		return fmt.Errorf("deleting SSHKey[%s] is failed: %s", key.ID, err)
+		return fmt.Errorf("deleting SSHKey[%s] is failed: %s", d.Id(), err)
 	}
 	return nil
 }

--- a/sakuracloud/resource_sakuracloud_vpc_router.go
+++ b/sakuracloud/resource_sakuracloud_vpc_router.go
@@ -588,12 +588,12 @@ func resourceSakuraCloudVPCRouterDelete(d *schema.ResourceData, meta interface{}
 
 	if vpcRouter.InstanceStatus.IsUp() {
 		if err := power.ShutdownVPCRouter(ctx, vrOp, zone, vpcRouter.ID, true); err != nil {
-			return fmt.Errorf("stopping VPCRouter[%s] is failed: %s", vpcRouter.ID, err)
+			return fmt.Errorf("stopping VPCRouter[%s] is failed: %s", d.Id(), err)
 		}
 	}
 
 	if err := vrOp.Delete(ctx, zone, vpcRouter.ID); err != nil {
-		return fmt.Errorf("deleting SakuraCloud VPCRouter[%s] is failed: %s", vpcRouter.ID, err)
+		return fmt.Errorf("deleting SakuraCloud VPCRouter[%s] is failed: %s", d.Id(), err)
 	}
 	return nil
 }

